src/main/java/internetshop/dao/jdbc/ShoppingCartJdbcImpl.java

```diff
@@ -0,0 +1,150 @@
+            statement.setLong(1, cart.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                cart.setId(resultSet.getLong(1));
+            }
+            setProducts(cart);
+            return cart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Shopping cart was not created", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * from shopping_carts where user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getShoppingCartFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cart was not got", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        deleteProducts(cart.getId());
+        setProducts(cart);
+        return cart;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteProducts(id);
+        String query = "DELETE FROM shopping_carts WHERE cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            if (statement.executeUpdate() > 0) {
+                return true;
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cart was not deleted", ex);
+        }
+        return false;
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        String query = "SELECT * FROM shopping_carts";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                shoppingCarts.add(getShoppingCartFromResultSet(resultSet));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Carts were not found", ex);
+        }
+        return shoppingCarts;
+    }
+
+    private ShoppingCart getShoppingCartFromResultSet(ResultSet rs) throws SQLException {
+        Long cartId = rs.getLong("cart_id");
+        Long userId = rs.getLong("user_id");
+        ShoppingCart shoppingCart = new ShoppingCart(cartId, userId);
+        shoppingCart.setProducts(getProducts(cartId));
+        return shoppingCart;
+    }
+
+    private ArrayList<Product> getProducts(Long cartId) {
```

Comment:
```suggestion
    private List<Product> getProducts(Long cartId) {
```
використовуй абстракцію, не імплементацію

Reply:
оу, точняк)